module-2/backend/week_two.md

```diff
@@ -25,20 +32,32 @@ end
 ```
 
 Now how would you find the owner of the team with an id of 4?
+Team.owners.find(4)
 
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+Not sure about this one
+
 6. Define foreign key, primary key, and schema.
-7. Describe the relationship between a foreign key on one table and a primary key on another table.
+A foreign key is a field in a table that identifes a row in another table. The row that the foreign key identifies is the primarh key of another table. A schema defines tables and indexes.
+7. Describe the relationship between a foreign key on one table and a primary key on another table. A foreign key in one table is the primary key in another table
 8. What are the parts of an HTTP response?
+Response,
+Status line / status code
+optional body
 
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+  .where(argument) will search through your table for all instances of row in your table that matches the argument being passed in the params.
+  .create will create and save an instance of a ruby object as a rown in your table
+  .find will find the first instance of a ruby object that matches the argumetn being passed in your params
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
 3. What two columns does `t.timestamps null: false` create in our database?
-4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
-5. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+:updated_at :created_at
+4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers? teachers :belongs_to school through: :classes
```

Comment:
Doesn't seem like this would need to be in a join. I would assume that a school had many teachers, and put a `school_id` on teachers unless we're assuming that teachers teach at many schools. Then I could potentially see the join you're describing.